addons/sourcemod/scripting/vsh/bosses/boss_uberranger.sp

```diff
@@ -0,0 +1,423 @@
+		boss.nClass = TFClass_Medic;
+		boss.iMaxRageDamage = 2500;
+		boss.bCanBeHealed = true;
+	}
+	
+	public void GetBossName(char[] sName, int length)
+	{
+		strcopy(sName, length, "White Über Ranger");
+	}
+	
+	public void GetBossInfo(char[] sInfo, int length)
+	{
+		StrCat(sInfo, length, "\nHealth: Very Low");
+		StrCat(sInfo, length, "\n ");
+		StrCat(sInfo, length, "\nAbilities");
+		StrCat(sInfo, length, "\n- Brave Jump (slower charge rate)");
+		StrCat(sInfo, length, "\n- Equipped with a Medi Gun");
+		StrCat(sInfo, length, "\n ");
+		StrCat(sInfo, length, "\nRage");
+		StrCat(sInfo, length, "\n- Übercharge for 6 seconds");
+		StrCat(sInfo, length, "\n- Summons a fellow Über Ranger");
+		StrCat(sInfo, length, "\n- Über Rangers are allowed to heal and über each other");
+		StrCat(sInfo, length, "\n- 200%% Rage: extends über duration to 8 seconds and summons 3 Über Rangers");
+	}
+	
+	public bool IsBossHidden()
+	{
+		return true;
+	}
+	
+	public void OnSpawn()
+	{
+		//Bosses and minions can't be overhealed, so a -max overheal attribute for the Medigun isn't needed
+		this.CallFunction("CreateWeapon", 211, "tf_weapon_medigun", 100, TFQual_Collectors, "");
+		
+		char sAttribs[128];
+		Format(sAttribs, sizeof(sAttribs), "2 ; 2.80 ; 252 ; 0.5 ; 259 ; 1.0"); 
+		int iWeapon = this.CallFunction("CreateWeapon", 173, "tf_weapon_bonesaw", 100, TFQual_Collectors, sAttribs);
+		if (iWeapon > MaxClients)
+			SetEntPropEnt(this.iClient, Prop_Send, "m_hActiveWeapon", iWeapon);
+			
+		/*
+		Vitasaw attributes:
+		
+		2: damage bonus
+		252: reduction in push force taken from damage
+		259: Deals 3x falling damage to the player you land on
+		*/
+		
+		//For reference: 230 230 230 is the color code for TF2's white paint
+		int iColor[4] = { 230, 230, 230, 255 };
+		
+		SetEntityRenderColor(this.iClient, iColor[0], iColor[1], iColor[2], iColor[3]);
+		
+		int iWearable = -1;
+		
+		//Interestingly, painting the cosmetics directly doesn't seem to make them white, so we paint it through attributes in this case
+		char sWhitePaint[16];
+		Format(sWhitePaint, sizeof(sWhitePaint), "142 ; 15132390"); 
+		
+		iWearable = this.CallFunction("CreateWeapon", 50, "tf_wearable", GetRandomInt(1, 100), TFQual_Normal, sWhitePaint);		//Prussian Pickelhaube
+		if (iWearable > MaxClients)
+		{
+			SetEntProp(iWearable, Prop_Send, "m_nModelIndexOverrides", g_iUberRangerPrussianPickelhaube);
+			//SetEntityRenderColor(iWearable, iColor[0], iColor[1], iColor[2], iColor[3]);
+		}
+		
+ 		iWearable = this.CallFunction("CreateWeapon", 315, "tf_wearable", GetRandomInt(1, 100), TFQual_Normal, sWhitePaint);	//Blighted Beak
+		if (iWearable > MaxClients)
+		{
+			SetEntProp(iWearable, Prop_Send, "m_nModelIndexOverrides", g_iUberRangerBlightedBeak);
+			//SetEntityRenderColor(iWearable, iColor[0], iColor[1], iColor[2], iColor[3]);
+		}
+			
+		for (int i = 1; i <= MaxClients; i++)
+			g_bUberRangerPlayerWasSummoned[i] = false;
+			
+		UberRangerResetColorList();
```

Comment:
Move this to `CUberRanger.CUberRanger` constructor, redundant to reset when boss suicides/respawns during pre-round